### PR TITLE
Support adjusting window transparency

### PR DESCRIPTION
**Pull request by astronomersiva**
This attempts to fix https://github.com/dawnlabs/carbon/issues/123. 

screen shot 2018-01-17 at 11 51 22 pm

### CODE REVIEW

File: components/Carbon.js [hunk 1]
```diff
@@ -178,7 +184,7 @@ class Carbon extends React.Component {
             }
 
             #container :global(.CodeMirror-gutters) {
-              background-color: unset;
+              background-color: unset !important;
```
Comment (astronomersiva) [inline components/Carbon.js]: some themes have `!important` in their background thereby rendering this useless.

File: lib/constants.js [hunk 2]
```diff
@@ -13,77 +13,95 @@ export const FONTS = [
 export const THEMES = [
   {
     id: '3024-night',
-    name: '3024 Night'
+    name: '3024 Night',
+    background: 'rgba(9, 3, 0, alpha) !important'
```
Comment (mfix22) [inline lib/constants.js]: I am thinking for this you can just look at the dom-node's background color dynamically and adjust the opacity as you have done here. That way if people want to add a new theme, they can do so a lot easier than manually setting an `rgba`.

Comment (astronomersiva) [reply lib/constants.js]: Makes sense. Will make the changes (y)

**Comment by astronomersiva** [comment]
Made some changes. Please have a look :)

File: components/Carbon.js [hunk 3]
```diff
@@ -99,6 +99,24 @@ class Carbon extends React.Component {
       viewportMargin: Infinity,
       lineWrapping: true
     }
+
+    let background = ''
+    let { codeMirrorNode } = this
+    if (codeMirrorNode) {
+      let [codeMirrorEditor] = codeMirrorNode.ref.children
+      let backgroundColor = getComputedStyle(codeMirrorEditor).backgroundColor
```
Comment (jakedex) [inline components/Carbon.js]: Are you sure this is working? It looks like the background color's always being set to black, regardless of syntax theme.

**Comment by ThiHaKo** [comment]
Nice

File: components/Carbon.js (hunk 3 above)
Comment (astronomersiva) [reply components/Carbon.js]: @jakedex, sorry for the late response.

I have made some changes and pushed. The problem with the earlier change was that when a theme was changed, the background calculated from the previous theme was still being applied.

I have addressed it in today's change but I am not able to reset the transparency slider once a new theme is selected. Any ideas?

**Comment by astronomersiva** [comment]
@jakedex, sorry for the late response.

I have made some changes and pushed. The problem with the earlier change was that when a theme was changed, the background calculated from the previous theme was still being applied.

I have addressed it in today's change but I am not able to reset the transparency slider once a new theme is selected. Any ideas?

**Pull request closed without merging**